ui/src/layouts/chats/presentation/messages/mod.rs

```diff
@@ -626,20 +626,33 @@ fn render_message<'a>(cx: Scope<'a, MessageProps<'a>>) -> Element<'a> {
                         .and_then(OsStr::to_str)
                         .map(str::to_string)
                         .unwrap_or_default();
-                    if let Some(file_path_to_download) = FileDialog::new()
-                    .set_directory(dirs::download_dir().unwrap_or_default()).set_file_name(&file_stem).add_filter("", &[&file_extension]).save_file() {
-                        let conv_id = message.inner.conversation_id();
-                        if !pending_downloads.read().contains_key(&conv_id) {
-                            pending_downloads.write().insert(conv_id, HashSet::new());
-                        }
-                        pending_downloads.write().get_mut(&conv_id).map(|conv| conv.insert(file.clone()));
+                    let conv_id = message.inner.conversation_id();
+
+                    if temp_dir.is_some() {
+                        let file_path_to_download = temp_dir.unwrap_or_default();
```

Comment:
We should be able to do a `if let` here instead
 
```suggestion
                    if let Some(file_path_to_download) = temp_dir {
```